checker.go

```diff
@@ -100,6 +100,18 @@ func (c *equalsChecker) Check(got interface{}, args []interface{}, note func(key
 			}
 		}
 	}
+
+	if reflect.ValueOf(got).Kind() == reflect.Ptr && reflect.ValueOf(want).Kind() == reflect.Ptr && Format(got) == Format(want) {
```

Comment:
Hmm, this isn't quite right when a custom formatter is used. I'm not sure what the right solution is there though.

Perhaps the answer is to move the logic into the printer, so it never prints as "same as" when the pointer values are the same, but instead prints something like `<same content as "foo" but with different pointer value>` or something?

WDYT?

Reply:
Good idea, done.